runtime/lua/vim/lsp/util.lua

```diff
@@ -1653,9 +1653,9 @@ function M.open_floating_preview(contents, syntax, opts)
     vim.wo[floating_winnr].concealcursor = 'n'
     vim.bo[floating_bufnr].filetype = 'markdown'
     vim.treesitter.start(floating_bufnr)
-    if not opts.height then
```

Comment:
This condition is still needed. If `opts.height` is specified then that's the exact height of the floating window that shouldn't be reduced.